src/api/oai/views.py

```diff
@@ -31,7 +32,10 @@ def oai_view_factory(request, *args, **kwargs):
             verb = DEFAULT_ENDPOINT
         elif verb not in ROUTES:
             raise exceptions.OAIBadVerb()
-        view = ROUTES.get(verb)
+        if request.repository:
+            view = PREPRINT_ROUTES.get(verb)
+        else:
+            view = ROUTES.get(verb)
```

Comment:
One issue with splitting the preprint routes from the previous views is that there won't be a unified feed from the press site anymore (preprints will only be returned from the OAI feed of the preprint site). This would be problematic for some installations where a publisher is hosting a combination of journals and preprints in the same Janeway instance.

Reply:
Thanks @mauromsl ! Yes, it will be great to have preprint and article merged and will make it a lot easier to create a unified feed.